fvm/evm/stdlib/contract.go

```diff
@@ -97,6 +99,55 @@ func (e abiDecodingError) Error() string {
 	return b.String()
 }
 
+func calculateComputation(
+	inter *interpreter.Interpreter,
+	values *interpreter.ArrayValue,
+	evmAddressTypeID common.TypeID,
+) uint {
+	computation := uint(0)
+
+	values.Iterate(inter, func(element interpreter.Value) (resume bool) {
+		switch value := element.(type) {
+		case *interpreter.StringValue:
+			// Dynamic variables, such as strings, are encoded
+			// in 3 chunks of 32 bytes. The first chunk contains
+			// the index where information for the string begin,
+			// the second chunk contains the number of bytes the
+			// string occupies, and the third chunk contains the
+			// value of the string itself.
+			computation += 2 * abiEncodingByteSize
+			stringLength := len(value.Str)
+			chunks := stringLength / abiEncodingByteSize
+			remainder := stringLength % abiEncodingByteSize
+			if remainder > 0 {
+				chunks += 1
+			}
+			computation += uint(chunks * abiEncodingByteSize)
+		case *interpreter.CompositeValue:
+			if value.TypeID() == evmAddressTypeID {
+				// EVM addresses are static variables with a fixed
+				// size of 32 bytes.
+				computation += abiEncodingByteSize
+			}
+		case *interpreter.ArrayValue:
+			// Dynamic variables, such as arrays and slices, encode
+			// an extra chunk of 32 bytes, for each element, except
+			// for the last one.
+			computation += uint((value.Count()-1)*abiEncodingByteSize) +
+				calculateComputation(inter, value, evmAddressTypeID)
+		default:
```

Comment:
Maybe refactor this to the explicitly supported types, and return an error for an unsupported type in the default case

Reply:
Updated in https://github.com/onflow/flow-go/pull/5146/commits/32e6b6f0b533f8de085671e5fff80c4e4587118f